src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,908 @@
+    };
+    window.addEventListener('message', this._eventListener);
+  }
+
+  destroy() {
+    if (this._eventListener) {
+      window.removeEventListener('message', this._eventListener);
+    }
+  }
+
+  async __getPage(increment: number) {
+    this.page += increment;
+
+    if (this.page <= 0) {
+      this.page = 1;
+    }
+    const from = (this.page - 1) * this.perPage;
+    const to = from + this.perPage;
+
+    await this.unlock();
+    let accounts;
+    if (this._isLedgerLiveHdPath()) {
+      accounts = await this._getAccountsBIP44(from, to);
+    } else {
+      accounts = this._getAccountsLegacy(from, to);
+    }
+    return accounts;
+  }
+
+  async _getAccountsBIP44(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const path = this._getPathForIndex(i);
+      const address = await this.unlock(path);
+      const valid = this.implementFullBIP44
+        ? await this._hasPreviousTransactions(address)
+        : true;
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+
+      // PER BIP44
+      // "Software should prevent a creation of an account if
+      // a previous account does not have a transaction history
+      // (meaning none of its addresses have been used before)."
+      if (!valid) {
+        break;
+      }
+    }
+    return accounts;
+  }
+
+  _getAccountsLegacy(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const address = this._addressFromIndex(pathBase, i);
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+      this.paths[ethUtil.toChecksumAddress(address)] = i;
+    }
+    return accounts;
+  }
+
+  _padLeftEven(hex: string) {
+    return hex.length % 2 === 0 ? hex : `0${hex}`;
+  }
+
+  _normalize(buf: Buffer) {
+    return this._padLeftEven(ethUtil.bufferToHex(buf).toLowerCase());
+  }
+
+  // eslint-disable-next-line no-shadow
+  _addressFromIndex(basePath: string, i: number) {
+    const dkey = this.hdk.derive(`${basePath}/${i}`);
+    const address = ethUtil
+      .publicToAddress(dkey.publicKey, true)
+      .toString('hex');
+    return ethUtil.toChecksumAddress(`0x${address}`);
+  }
+
+  _pathFromAddress(address: string) {
+    const checksummedAddress = ethUtil.toChecksumAddress(address);
+    let index = this.paths[checksummedAddress];
+    if (typeof index === 'undefined') {
+      for (let i = 0; i < MAX_INDEX; i++) {
+        if (checksummedAddress === this._addressFromIndex(pathBase, i)) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (typeof index === 'undefined') {
+      throw new Error('Unknown address');
+    }
+    return this._getPathForIndex(index);
+  }
+
+  _toAscii(hex: string) {
```

Comment:
This method is unused and is marked as private, we could potentially remove it, but maybe this is out of the PRs scope?